### PR TITLE
BaSP-MR-153: React hook form implementation - Trainers

### PR DESCRIPTION
**Pull request by EmaBaldor**
Trainers: Update of the form, using 'react-hook-form' and implementation of the 'input' shared component. 
In addition, the corresponding validations were added.

### CODE REVIEW

**Comment by EmaBaldor** [comment]
[[Trainers] - React-hook-form implementation](https://trello.com/c/jtmRRsHd/153-trainers-react-hook-form-implementation)

File: src/Components/Trainers/Form/index.jsx [hunk 1]
```diff
@@ -15,86 +19,68 @@ const Form = () => {
   const [shouldRedirect, setShouldRedirect] = useState(false);
   const dispatch = useDispatch();
   const trainers = useSelector((state) => state.trainers.data);
```
Comment (fernandofernandezuy) [inline src/Components/Trainers/Form/index.jsx:21]: I think you could avoid getting all the trainers for this form, you could use a getById instead

File: src/Components/Trainers/Form/index.jsx [hunk 2]
```diff
@@ -109,96 +95,97 @@ const Form = () => {
...
-      <div className={styles.buttons}>
-        <Button text="Cancel" type="cancel" clickAction={handleCancel} />
-        <Button text={id ? 'Confirm' : 'Submit'} type="submit" clickAction={handleSubmit} />
+      <div>
+        <div className={styles.buttons}>
+          <Button text="Back" type="cancel" clickAction={handleCancel} />
+          <Button text={id ? 'Update' : 'Add'} type="submit" />
+        </div>
+        <div className={styles.buttons}>
+          <Button type={'cancel'} clickAction={handleReset} text={'Reset'} />
```
Comment (fernandofernandezuy) [inline src/Components/Trainers/Form/index.jsx]: Your reset button its not working properly, when i click it does the post or put method also.

**Review by fernandofernandezuy — CHANGES_REQUESTED** [review]

**Review by fernandofernandezuy — APPROVED** [review]

**Review by ale-schmidt — COMMENTED** [review]
Please specify the name of the entity you are working on in the title of the PR

**Review by ale-schmidt — APPROVED** [review]
All working and looking fine :)

**Comment by nazarenapantanetti** [comment]
Tested in master, changes requested! Follow the link to see the full report: https://trello.com/c/ntFGT8UP/182-trainer-bug-validations-fails

**Pull request merged**